autogpt_platform/backend/backend/blocks/system/library_operations.py

```diff
@@ -0,0 +1,316 @@
+        store_listing_version_id: str = SchemaField(
+            description="The ID of the store listing version to add to library"
+        )
+        agent_name: Optional[str] = SchemaField(
+            description="Optional custom name for the agent in your library",
+            default=None,
+        )
+
+    class Output(BlockSchema):
+        success: bool = SchemaField(
+            description="Whether the agent was successfully added to library"
+        )
+        library_agent_id: str = SchemaField(
+            description="The ID of the library agent entry"
+        )
+        agent_id: str = SchemaField(description="The ID of the agent graph")
+        agent_version: int = SchemaField(
+            description="The version number of the agent graph"
+        )
+        agent_name: str = SchemaField(description="The name of the agent")
+        message: str = SchemaField(description="Success or error message")
+
+    def __init__(self):
+        super().__init__(
+            id="2602a7b1-3f4d-4e5f-9c8b-1a2b3c4d5e6f",
+            description="Add an agent from the store to your personal library",
+            categories={BlockCategory.BASIC},
+            input_schema=AddToLibraryFromStoreBlock.Input,
+            output_schema=AddToLibraryFromStoreBlock.Output,
+            test_input={
+                "store_listing_version_id": "test-listing-id",
+                "agent_name": "My Custom Agent",
+            },
+            test_output=[
+                ("success", True),
+                ("library_agent_id", "test-library-id"),
+                ("agent_id", "test-agent-id"),
+                ("agent_version", 1),
+                ("agent_name", "Test Agent"),
+                ("message", "Agent successfully added to library"),
+            ],
+            test_mock={
+                "_add_to_library": lambda *_, **__: LibraryAgent(
+                    library_agent_id="test-library-id",
+                    agent_id="test-agent-id",
+                    agent_version=1,
+                    agent_name="Test Agent",
+                )
+            },
+        )
+
+    async def run(
+        self,
+        input_data: Input,
+        *,
+        user_id: str,
+        **kwargs,
+    ) -> BlockOutput:
+        try:
+            library_agent = await self._add_to_library(
+                user_id=user_id,
+                store_listing_version_id=input_data.store_listing_version_id,
+                custom_name=input_data.agent_name,
+            )
+
+            yield "success", True
+            yield "library_agent_id", library_agent.library_agent_id
+            yield "agent_id", library_agent.agent_id
+            yield "agent_version", library_agent.agent_version
+            yield "agent_name", library_agent.agent_name
+            yield "message", "Agent successfully added to library"
+
+        except store_exceptions.AgentNotFoundError as e:
+            logger.warning(f"Agent not found: {str(e)}")
```

Comment:
Exact same handling on both cases, we should collapse them and do an if-else for the message.

though what's the point fo doing this, why can't we just let it raise an exception and has it yield "error" and avoid producing bogus output.